torch_xla_py/xla_dist.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+"""Tool to distribute training on Cloud TPU Pods."""
+from __future__ import division
+from __future__ import print_function
+
+
+class Worker(object):
+
+  def __init__(self, internal_ip, mach_type, zone, hostname=None):
+    self._internal_ip = internal_ip
+    self._mach_type = mach_type
+    self._zone = zone
+    self._hostname = hostname
+
+
+class Cluster(object):
+
+  def __init__(self, workers):
```

Comment:
This might be in a followup PR: we should be able to init a `Cluster`(VM) from an instance group name right? For users this can be one line instead of setting up each worker manually. :D

Reply:
Oh yeah, so initializing the `Cluster` with `Worker`s by fetching the instance group metadata and calling GCE will all be done by the `ClusterResolver`. I actually have it all done and will send out after a PR or two. Thanks!

Reply:
Hmmm could you explain what's the difference between `Cluster/ClusterResolver`? If we are going to have another layer of `DistributedExecutor`, I'm not sure I fully understand the difference between layers :P